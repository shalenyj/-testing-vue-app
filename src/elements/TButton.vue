<template>
  <button :type="type" class="t-button" @click="handleClick">
    {{ title }}
  </button>
</template>

<script>
export default {
  name: "TButton",
  props: {
    title: {
      type: String,
      required: true,
    },
    type: {
      type: String,
      default: "button",
    },
  },
  methods: {
    handleClick() {
      this.$emit("click");
    },
  },
};
</script>

<style lang="css" scoped>
.t-button {
  background: cadetblue;
  color: white;
  border: 2px solid lightblue;
  border-radius: 4px;
  padding: 5px 10px;
  text-transform: capitalize;
}
</style>
