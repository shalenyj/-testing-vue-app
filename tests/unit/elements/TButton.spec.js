import { mount } from "@vue/test-utils";

import TButton from "@/elements/TButton";

//Snapshot

describe("@/elements/TButton", () => {
  it("should mount with title", () => {
    const wrapper = mount(TButton, {
      propsData: {
        title: "Hey now",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should mount with not default type", () => {
    const wrapper = mount(TButton, {
      propsData: {
        title: "Hey now",
        type: "submit",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
