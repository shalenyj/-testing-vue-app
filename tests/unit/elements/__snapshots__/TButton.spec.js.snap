// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/elements/TButton should mount with not default type 1`] = `
<button
  class="t-button"
  type="submit"
>
  
  Hey now

</button>
`;

exports[`@/elements/TButton should mount with title 1`] = `
<button
  class="t-button"
  type="button"
>
  
  Hey now

</button>
`;
